main.go

```diff
@@ -125,3 +141,38 @@ func main() {
 		klog.Fatalf("unable to run the manager: %v", err)
 	}
 }
+
+// createClientConfigs allow users to provide second config using management-kubeconfig, if specified
+// try to build it from provided path. First returned value is management config used for Machines,
+// second is workload config used for Node/CSRs.
+func createClientConfigs(managementClientPath string, managementConfigAuthoritative bool) (*rest.Config, *rest.Config, error) {
+	managementConfig := control.GetConfigOrDie()
```

Comment:
If I'm reading this correctly the workloadConfig is always in-cluster?
We need the workloadConfig to be configurable. We want to run the machine approver in the management cluster

Also I know @JoelSpeed suggested the `managementConfigAuthoritative` pattern but I can't recall why is that necessary?

Reply:
oh I see this would use --kubeconfig if set.
May be is only me but this pattern results very confusing.
Why the need for the bool managementConfigAuthoritative?

Reply:
I think there were several cases we were trying to account for

1. Cluster is a joined cluster, in-cluster config for both, no flags set
2. CMA is running on management, `management-kubeconfig` should be in-cluster, `kubeconfig` should be set to workload cluster
3. CMA is running in workload, `management-kubeconfig` should be set to management cluster, `kubeconfig` should be in-cluster
4. You are running CMA remotely, `kubeconfig` is set to a kubeconfig which then both clients use

I think for number 2 and number 4 to work (this was discussed in CA conversations upstream), we assume that if `kubeconfig != "" && managementKubeconfig == ""` then `kubeconfig` should be used for all clients, the authoritative flag prevents that assumption IIRC?

Reply:
This duplicates autoscaler behaviour 
https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/clusterapi#connecting-cluster-autoscaler-to-cluster-api-management-and-workload-clusters  

https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_provider.go#L139

Reply:
The motivation for the bool pattern on the autoscaler was to keep a fallback to previous behaviour by default when --cloud-config is not specified for backward compatibility.
I don't think we need to keep backward compatibility for these flags here since this is a self manage internal component.

Is there a reason we can't just have `--management-cluster-kubeconfig` and `--workload-cluster-kubeconfig`, that seems the most intuitive and should cover all cases?

wdyt?

Reply:
That seems fine to me, will just be one extra flag to know about when developing locally, not a major problem really. This should affect IPI clusters as they will both default to in-cluster config so happy with that if it's acceptable for the HyperShift project